src/components/shared/DataTypeSelect.vue

```diff
@@ -0,0 +1,54 @@
+<template>
```

Comment:
I decided to part with the pattern in LibraryTypeSelect, and decided that the label should be managed independently. We should ideally adopt the same pattern in LibraryTypeSelect.

This separates out the field from its wider context, increasing reusability. (For example I can use it in auto-generated forms, or decide I need to put the label alongside the field, rather than above it)

I noticed that the libraryType select had already been picking up lots of configuration for this kind of thing.